docs/providers/aws/guide/testing.md

```diff
@@ -77,22 +81,30 @@ class Users {
   });
   }
 }
-```
-
-This class is testable and doesn't require running any of the external services. Instead of real `db` and `mailer` objects we can pass mocks and assert if `saveUser` and `sendWelcomeEmail` has been called with proper arguments. We should have as much unit-test as possible and run them every code change. Of course passing unit-tests doesn't mean that our function is working as expected. That's why we also need integration test.
 
-## Integration tests
-
-After extracting business logic to separate module handler function looks like this:
+module.exports = Users;
+```
 
 ```javascript
 const db = require('db').connect();
 const mailer = require('mailer');
-const users = require('users')(db, mailer);
+const Users = require('users');
+
+let users = new Users(db, mailer);
 
 module.exports.saveUser = (event, context, callback) => {
   users.save(event.email, callback);
 };
 ```
 
-It's responsible for setting up dependencies, injecting them and calling business logic functions. This code will be changed less often. To make sure that function is working as expected integration tests should be ran against deployed function. They should invoke function (`serverless invoke`) with fixture email address, check if user is actually saved to DB and check if email was received.
+Now, the above class keeps business logic separate.  Further, the code responsible for setting up dependencies, injecting them, calling business logic functions and interacting with AWS Lambda is in its own file, which will be changed less often.  This way, the business logic is not provider dependent, easier to re-use, and easier to test.
+
+Further, this code doesn't require running any external services.  Instead of a real `db` and `mailer` services, we can pass mocks and assert if `saveUser` and `sendWelcomeEmail` has been called with proper arguments.
+
+Unit Tests can easily be written to cover the above class.  Then invoke the function (`serverless invoke`) with fixture email address, check if user is actually saved to DB and check if email was received to see if everything is working together.
```

Comment:
IMHO it's worth to add that `serverless invoke` is helpful "integration tests". I mean we should use "integration tests" phrase :)